aexpect/session_ops.py

```diff
@@ -0,0 +1,598 @@
+# This Python file uses the following encoding: utf-8
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+
+"""
+
+SYNOPSIS
+------------------------------------------------------
+A list of frequent operations performed through remote sessions.
+
+Copyright: 2018 Intra2net AG
+
+
+DESCRIPTION
+------------------------------------------------------
+Most commands passed to a session are routine invocations of common userland
+tools. Nevertheless some of them require a certain effort to set up because
+they are squeezed through the eye of the needle, the shell session.
+
+The functionality in this module assumes a guest with the ordinary userland
+tools available. Some of it might just work on Windows guests if the proper
+environment (i. e. Cygwin) is present.
+
+This module aims to provide convenient wrappers for butter-and-bread
+functionality on a Linux system. Currently, the implemented functions fall
+under four categories: ``grep``, ``stat``, ``test``, and common file ops.
+
+grep:
+Standard case search operations. It is always assumed that binaries in the path
+should be searched (the ``-a`` flag).
+
+stat:
+Allows passing an optional format argument (``-c``). Custom wrappers for
+extracting *atime*, *ctime*, and *mtime*, etc exist.
+
+test:
+Allows for testing existence, permissions and other properties of files.
+
+file ops:
+Simple file operations executed through sessions. Most functions boil down to
+executing simple commands like ls, tar, md5sum, etc. If these operations return
+non-0, a :py:class:`RuntimeError` is raised, containing the commands's error
+message. All functions :py:func:`shlex.quote` their args for better security.
+
+
+IMPLEMENTATION
+------------------------------------------------------
+
+"""
+
+import sys
+import re
+from os.path import join, split as path_split
+
+if sys.version_info[0] < 3:
+    from six.moves import shlex_quote as quote
+else:
+    from shlex import quote   # pylint: disable=E0611
```

Comment:
Not sure here, but maybe I would try to import the `shlex` version and in case of failures try the `six.moves`:

```
try:
  from shlex import quote
except ImportError:
  from six.moves import shlex_quote as quote
```

Reply:
I would agree but we no longer support python 2 so this is no longer of importance. See the recent push for details.

Reply:
That is one reason because I and other developers prefer a squashed version. I'm reviweing an old code. ;)

Reply:
Again, reviewing old code is your own choice. Also, this PR has been waiting for about a year now so I am referring to python 2 being dropped from aexpect just a few weeks ago.